internal/backend/runtime/omni/migration/helpers.go

```diff
@@ -30,7 +30,7 @@ func reconcileConfigInputs(ctx context.Context, st state.State, item *omni.Clust
 		return err
 	}
 
-	// update input versions on the cluster machine config to avoid it's reconciliation
+	// update input versions on the cluster machine config to avoid its reconciliation
```

Comment:
As you dropped `ClusterMachineTalosVersion` from the config inputs you also need to drop it here.